compiler/rustc_infer/src/infer/error_reporting/mod.rs

```diff
@@ -1507,7 +1507,7 @@ impl<'tcx> TypeErrCtxt<'_, 'tcx> {
             (ty::FnPtr(sig1), ty::FnPtr(sig2)) => self.cmp_fn_sig(sig1, sig2),
 
             _ => {
-                if t1 == t2 && !self.tcx.sess.verbose() {
+                if t1 == t2 && !self.tcx.sess.opts.verbose {
                     // The two types are the same, elide and don't highlight.
                     (DiagnosticStyledString::normal("_"), DiagnosticStyledString::normal("_"))
                 } else {
```

Comment:
Just noticed that this now needs an else branch for the case where verbose is on and t1 != t2, to print out the type but with a normal style, otherwise we are going to be highlighting a lot more than we'd want.